Assets/Plugins/Krearthur/GOPainter/Scripts/GOPainter.cs

```diff
@@ -1837,67 +1633,46 @@ protected void HandleShiftScroll(Event e)
             {
                 float degreesBefore = circleFactory.degrees;
                 circleFactory.degrees += e.delta.y < 0 ? 10 : -10;
-                if (circleFactory.degrees >= 360) circleFactory.degrees = 360;
-                if (circleFactory.degrees <= 0) circleFactory.degrees = 0;
+                if (circleFactory.degrees >= 360)
+                    circleFactory.degrees = 360;
+                if (circleFactory.degrees <= 0)
+                    circleFactory.degrees = 0;
                 float degreesAfter = circleFactory.degrees;
-                if (degreesBefore != degreesAfter) PlaySoundRotate();
-                if (isDragging) circleFactory.Produce();
+                if (degreesBefore != degreesAfter)
+                    audio.PlaySoundRotate();
+                if (isDragging)
+                    circleFactory.Produce();
 
             }
             else
             {// set canvas position
-
                 if (canvasAxis == CanvasAxis.X)
-                {
-                    if (e.delta.y < 0 && canvasPositionX < paintCanvasSize - 1)
-                    {
-                        PlaySoundMovePlaneUp();
-                        canvasPositionX += gridSize.x;
-                        if (canvasPositionX >= paintCanvasSize / 2) canvasPositionX = paintCanvasSize / 2 - 1;
-                    }
-                    else if (e.delta.y > 0 && canvasPositionX > -paintCanvasSize / 2)
-                    {
-                        PlaySoundMovePlaneDown();
-                        canvasPositionX -= gridSize.x;
-                        if (canvasPositionX < -paintCanvasSize / 2) canvasPositionX = -paintCanvasSize / 2;
-                    }
-                }
+                    SetCanvasPosition(ref canvasPositionX, gridSize.x);
                 else if (canvasAxis == CanvasAxis.Y)
+                    SetCanvasPosition(ref canvasPositionY, gridSize.y);
+                else if (canvasAxis == CanvasAxis.Z)
+                    SetCanvasPosition(ref canvasPositionZ, gridSize.z);
+                UpdatePaintCanvasSize();
+            }
+            e.Use();
+
+            void SetCanvasPosition(ref float canvasPosition, float gridSize)
```

Comment:
The repeatable logic is placed in a separate `void SetCanvasPosition(ref float canvasPosition, float gridSize)` method, inside which the logic is simplified for greater understanding.